dkg/disk.go

```diff
@@ -112,10 +116,74 @@ func writeDepositData(aggSigs map[core.PubKey]*bls_sig.Signature, withdrawalAddr
 
 	// Write it to disk
 	depositPath := path.Join(dataDir, "deposit-data.json")
-	err = os.WriteFile(depositPath, bytes, 0o400) // read-only
+	err = os.WriteFile(depositPath, bytes, 0o444) // read-only
 	if err != nil {
 		return errors.Wrap(err, "write deposit data")
 	}
 
 	return nil
 }
+
+// checkWrites writes sample files to check disk writes and removes sample files after verification.
+func checkWrites(dataDir string, def cluster.Definition) error {
+	var shares []share
+	sigs := make(map[core.PubKey]*bls_sig.Signature)
+	for i := 0; i < def.NumValidators; i++ {
+		tss, sks, err := tbls.GenerateTSS(def.Threshold, len(def.Operators), rand.Reader)
+		if err != nil {
+			return err
+		}
+
+		shares = append(shares, share{
+			PubKey:       tss.PublicKey(),
+			SecretShare:  sks[0],
+			PublicShares: tss.PublicShares(),
+		})
+
+		pk, err := tblsconv.KeyToCore(tss.PublicKey())
+		if err != nil {
+			return err
+		}
+
+		sig, err := testutil.RandomBLSSignature()
+		if err != nil {
+			return err
+		}
+
+		sigs[pk] = sig
+	}
+
+	if err := writeDepositData(sigs, "0x0000000000000000000000000000000000000000", "prater", dataDir); err != nil {
+		return err
```

Comment:
Are these error message human friendly? Will people understand what the problem is?